File: typescript/src/PlanetTS1/Model/Player/Buildings/PNRGPlant.ts
import { ALocation as ALocation } from "../../Utilities"

export class PNRGPlant extends NRGPlant {
    loc : ALocation
    use : boolean
    owner : number
    icon : string

    constructor(location : ALocation) {
        super()
        this.loc = location
        this.use = true
        this.owner = 1
        this.icon = "pP"
    }
}

File: dotnet/src/Model/Player/Buildings/PNRGPlant.cs
using System;

namespace PlanetCS1 {
    class PNRGPlant : NRGPlant {
        public PNRGPlant(Location location) {
            this.loc = location;
            this.use = true;
            this.owner = 1;
            this.icon = "pP";
        }
    }
}